lib/xchain/types.go

```diff
@@ -248,6 +249,24 @@ func (a Attestation) AttestationRoot() ([32]byte, error) {
 	return AttestationRoot(a.AttestHeader, a.BlockHeader, a.MsgRoot)
 }
 
+// Verify verifies all the signatures in the attestation against the sigtuple validator address.
```

Comment:
nit: don't repeat the logic of the function of the comments. It will just get out of date. Rather focus on the return types.

e.g. `Verify returns an error if the attestation is invalid.`